corehq/apps/analytics/static/analytics/js/logging.js

```diff
@@ -0,0 +1,176 @@
+/* globals _, JSON */
+/**
+ * To Use this library, enable the feature flag ANALYTICS_NEW.
+ *
+ * To enable logging, enable the following INTERNAL feature flags to see each level:
+ * DEBUG: ANALYTICS_DEBUG
+ * VERBOSE: ANALYTICS_VERBOSE
+ * WARNING, DEPRECATED: ANALYTICS_DEPRECATED
+ *
+ */
+hqDefine('analytics/js/logging', function () {
+    'use strict';
+    var _LEVELS = {
+        warning: ['WARNING', 'color: #994f00;'],
+        verbose: ['VERBOSE', 'color: #685c53;'],
+        debug: ['DEBUG', 'color: #004ebc;'],
+        deprecated: ['DEPRECATED', 'color: #e53e30;'],
+    };
+
+    var _printPretty = function (message) {
+        var _title, _value, _grp;
+        if (_.isUndefined(message.value)) {
+            console.log("Message was undefined");  // eslint-disable-line no-console
+        } else if (message.value.isGroup) {
+            _grp = message.value;
+            _grp.level = message.level;
+            _grp.print();
+        } else if (_.isArguments(message.value)) {
+            _title = "Arguments (" + message.value.length + ")    " + _.map(Array.from(message.value), JSON.stringify).join('    ');
+            _value = Array.from(message.value);
+            _grp = new Group(_title, new Message(_value, message.level));
+            _grp.isCollapsed = true;
+            _grp.print();
+        } else if (_.isArray(message.value)) {
+            _.each(message.value, function (msg) {
+                _printPretty(new Message(msg, message.level));
+            });
+        } else if (_.isObject(message.value) && _.has(message.value, 0) && _.isElement(message.value[0])) {
+            // DOM element
+            _title = "#" + message.value.get(0).id + " ." + message.value.get(0).className.split(' ').join('.');
+            _value = message.value.get(0).outerHTML;
+            _grp = new Group(_title, new Message(_value, message.level));
+            _grp.isCollapsed = true;
+            _grp.print();
+        } else if (!_.isString(message.value) && !_.isUndefined(message.value)) {
+            _title = JSON.stringify(message.value);
+            _value = message.value;
+            _grp = new Group(_title, new Message(_value, message.level));
+            _grp.isCollapsed = true;
+            _grp.isRaw = true;
+            _grp.print();
+        } else {
+            message.print();
+        }
+    };
+
+    var _getStyle = function (level) {
+        var levelOptions = _LEVELS[level];
+        return (levelOptions) ? levelOptions[1] : "";
+    };
+
+    var Message = function (value, level) {
+        var msg = {};
+        msg.level = level;
+        msg.value = value;
+        msg.print = function () {
+            console.log(msg.value);  // eslint-disable-line no-console
```

Comment:
Mentioned this elsewhere, but after looking a bit at the [MDN docs](https://developer.mozilla.org/en-US/Add-ons/SDK/Tools/console), I'm even more in favor of taking advantage of the built-in `debug`, `info`, etc. That would also allow you to get rid of the `ANALYTICS_DEBUG`, `ANALYTICS_WARNING`, and `ANALYTICS_VERBOSE` feature flags; people could just interact normally with the browser's web console logging controls.

Reply:
It's nice to have at least one feature flag to disable all `console` statements from analytics altogether

Reply:
I said this above, but I've discovered that unfortunately you can't apply the console levels to groups. also it feels like a good idea to completely turn off any console output altogether